src/cdk/commands/aslVisualizationCDK.ts

```diff
@@ -0,0 +1,122 @@
+    }
+
+    protected override setupWebviewPanel(uniqueIdentifier: string): vscode.WebviewPanel {
+        const logger: Logger = getLogger()
+
+        // Create and show panel
+        const panel = this.createVisualizationWebviewPanel(uniqueIdentifier)
+
+        // Set the initial html for the webpage
+        panel.webview.html = this.getWebviewContent(
+            panel.webview.asWebviewUri(ext.visualizationResourcePaths.webviewBodyScript),
+            panel.webview.asWebviewUri(ext.visualizationResourcePaths.visualizationLibraryScript),
+            panel.webview.asWebviewUri(ext.visualizationResourcePaths.visualizationLibraryCSS),
+            panel.webview.asWebviewUri(ext.visualizationResourcePaths.stateMachineCustomThemeCSS),
+            panel.webview.cspSource,
+            {
+                inSync: localize(
+                    'AWS.stepFunctions.graph.status.inSync',
+                    'Previewing ASL state machine'
+                ),
+                notInSync: localize('AWS.stepFunctions.graph.status.notInSync', 'Errors detected. Cannot preview.'),
+                syncing: localize('AWS.stepFunctions.graph.status.syncing', 'Rendering ASL graph...'),
+            }
+        )
+
+        const debouncedUpdate = debounce(this.sendUpdateMessage.bind(this), 500)
+
+        // Handle messages from the webview
+        this.disposables.push(
+            panel.webview.onDidReceiveMessage(async (message: MessageObject) => {
+                switch (message.command) {
+                    case 'updateResult':
+                        logger.debug(message.text)
+                        if (message.error) {
+                            logger.error(message.error)
+                        }
+                        break
+                    case 'webviewRendered': {
+                        // Webview has finished rendering, so now we can give it our
+                        // initial state machine definition.
+                        await this.sendUpdateMessage(this.cfnDefinition)
+                        break
+                    }
+                }
+            })
+        )
+
+        // When the panel is closed, dispose of any disposables/remove subscriptions
+        const disposePanel = () => {
+            if (this.isPanelDisposed) {
+                return
+            }
+            this.isPanelDisposed = true
+            debouncedUpdate.cancel()
+            this.onVisualizationDisposeEmitter.fire()
+            this.disposables.forEach(disposable => {
+                disposable.dispose()
+            })
+            this.onVisualizationDisposeEmitter.dispose()
+        }
+
+        this.disposables.push(
+            panel.onDidDispose(() => {
+                disposePanel()
+            })
+        )
+
+        return panel
+    }
+
+    protected override makeWebviewTitle(uniqueIdentifier: string): string {
+        return localize('AWS.stepFunctions.graph.titlePrefix', 'Graph: {0}', uniqueIdentifier)
+    }
```

Comment:
We should confirm the behaviour if the user deletes the `template.json` file then re-runs `cdk synth`.

Reply:
I changed it so that if user deletes template.json, the webview panel get closed, and will show an error message when user tries to render the graph again. If user re-runs ```cdk synth``` and pushes the render icon button, graph gets rendered again.